components/shapediver/parameter/select/SelectComponent.tsx

```diff
@@ -0,0 +1,76 @@
+import {MantineColor} from "@mantine/core";
+import React from "react";
+import SelectButtonFlexComponent from "./SelectButtonFlexComponent";
+import SelectButtonGroupComponent from "./SelectButtonGroupComponent";
+import SelectChipGroupComponent from "./SelectChipGroupComponent";
+import SelectColorComponent from "./SelectColorComponent";
+import SelectDropDownComponent from "./SelectDropDownComponent";
+import SelectFullWidthCards from "./SelectFullWidthCards";
+import SelectImageDropDownComponent from "./SelectImageDropDownComponent";
+
+export interface SelectComponentItemDataType {
+	/** Display name to use instead of the item name. */
+	displayname?: string;
+	/** Tooltip. */
+	tooltip?: string;
+	/** Description. */
+	description?: string;
+	/** URL to image. Can be a data URL including a base 64 encoded image. */
+	imageUrl?: string;
+	/** Optional color, used for color selection components. */
+	color?: MantineColor;
+	/** Optional width. */
+	width?: number;
+}
+
+export interface SelectComponentProps {
+	/** Controlled value (item name). */
+	value: string | null | undefined;
+	/** Handler for updating the value. */
+	onChange: (value: string | null) => void;
+	/** Item names that can be selected, must be unique. */
+	items: string[];
+	/** Record containing optional further item data per item name. */
+	itemData?: Record<string, SelectComponentItemDataType>;
+	/** Whether the component shall be disabled. */
+	disabled?: boolean;
+}
```

Comment:
@LemToUp I suggest to introduce a further property called `settings` which can be used to pass settings that are specific to the component, like the image `width` for the SelectImageDropDownComponent

Reply:
Replaced: https://github.com/shapediver/AppBuilderShared/pull/77/commits/7c57aeba8fecb6c50269ca31ee057b7b409d8113